eng/dev_requirements.txt

```diff
@@ -1,8 +1,12 @@
+pyright==1.1.389
+pylint==3.2.7
+tox==4.16.0
+mypy==1.13.0
 azure-pylint-guidelines-checker==0.0.8
 colorama==0.4.6
-debugpy==1.8.9
-pytest==8.3.4
-tox==4.23.2
-coverage==7.6.9
-black==24.10.0
-setuptools==73.0.1
+debugpy==1.8.2
+pytest==8.3.2
+coverage==7.6.1
+black==24.8.0
+ptvsd==4.3.2
+types-PyYAML==6.0.12.8
```

Comment:
This file shall be synced with https://github.com/microsoft/typespec/blob/main/packages/http-client-python/generator/dev_requirements.txt